scriptmodules/ports/hexen2.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+# This file is part of The RetroPie Project
+#
+# The RetroPie Project is the legal property of its developers, whose names are
+# too numerous to list here. Please refer to the COPYRIGHT.md file distributed with this source.
+#
+# See the LICENSE.md file at the top-level directory of this distribution and
+# at https://raw.githubusercontent.com/RetroPie/RetroPie-Setup/master/LICENSE.md
+#
+
+rp_module_id="hexen2"
+rp_module_desc="Hexen II - Hammer of Thyrion source port"
+rp_module_licence="GPL2 https://raw.githubusercontent.com/svn2github/uhexen2/master/docs/COPYING"
+rp_module_help="For registered version, please add your full version PAK files to $romdir/ports/hexen2/data1/ to play. These files for the registered version are required: pak0.pak, pak1.pak and strings.txt. The registered pak files must be patched to 1.11 for Hammer of Thyrion."
+rp_module_section="exp"
+rp_module_flags=""
+
+function depends_hexen2() {
+    getDepends cmake libsdl1.2-dev libsdl-net1.2-dev libsdl-sound1.2-dev libsdl-mixer1.2-dev libsdl-image1.2-dev timidity
+}
+
+function sources_hexen2() {
+    gitPullOrClone "$md_build" https://github.com/svn2github/uhexen2.git
+}
+
+function build_hexen2() {
+    cd "$md_build/engine/hexen2"
+    ./build_all.sh
+    md_ret_require="$md_build/engine/hexen2/hexen2"
+}
+
+function install_hexen2() {
+    md_ret_files=(
+       'engine/hexen2/hexen2'
+    )
+}
+
+function game_data_hexen2() {
+    if [[ ! -f "$romdir/ports/hexen2/data1/pak0.pak" ]]; then
+        downloadAndExtract "https://netix.dl.sourceforge.net/project/uhexen2/Hexen2Demo-Nov.1997/hexen2demo_nov1997-linux-i586.tgz" "$romdir/ports/hexen2" --strip-components 1 "hexen2demo_nov1997/data1"
+    fi
+}
+
+function configure_hexen2() {
+    addPort "$md_id" "hexen2" "Hexen II" "$md_inst/hexen2"
+
+    mkRomDir "ports/hexen2"
+
+    moveConfigDir "$home/.hexen2" "$romdir/ports/hexen2"
```

Comment:
Actually, I'd like to just check this before merging. I'd prefer this directory wasn't linked to the ROM ports folder. Better if possible to link to config area, but pass parameters to load data files from.right location (if possible).

Reply:
The binary is expecting that the game data is located in $home/.hexen2 and when executing the hexen2 binary it looks at it's configdir for a data1 folder with the game data files ex. pak0.pak.

I'm not sure which other parameter can be used instead of moving/symlinking the $configdir to the $romdir. Maybe it could be simplified like the enhanced version of `downloadAndExtract`.

Reply:
@joolswills I checked the source code now and indeed it is hard-coded. See here:
https://github.com/svn2github/uhexen2/blob/1f6c371faa563db30a2e26909bb8ea49ceefcda4/engine/h2shared/userdir.h#L26-L49

I see three solutions (in descending order of difficulty):
1. Symlink `$HOME/.hexen2` to `$romdir/ports/hexen2` and keep the data files there (current solution).
2. Make a patch in to change the hard-coded path in the source.
3. Make a patch in to add a `--userdir` option to the program.

Option 2) is easy but I don't like the idea of hard-coding `$romdir/ports/hexen2` either.
Option 3) is the ideal case as you mention but requires more effort. I'm not sure it's worth it to do here, i.e. should be done upstream actually.

I personally don't think there is anything wrong with Option 1) as the actual files are kept in `$romdir/ports/hexen2` despite the unfortunate decision from upstream to statically look for these data files in a config directory, e.g. `$HOME/.hexen2`. The symlink approach is not that ugly and in my humble opinion doesn't fully break RP's design.

---

> I'm not sure which other parameter can be used instead of moving/symlinking the $configdir to the $romdir. Maybe it could be simplified like the enhanced version of `downloadAndExtract`.

With current source code, nothing can be done except patching the program :)

Reply:
Great work @hhromic 

I know that symlinking `$HOME/.hexen2` to `$romdir/ports/hexen2` is not the most ideal way but creating this symlink is to make it easier to replace the game data files for users of RetroPie. Otherwise users will not have the ease of replacing the game data files since it would require them to go into $home/.hexen2.

For Option 3, wouldn't that require making a patch and creating a pull request into the source ports GIT repository in order for that to be accomplished?

Reply:
> For Option 3, wouldn't that require making a patch and creating a pull request into the source ports GIT repository in order for that to be accomplished?

It is not "required" but highly desirable. We can always add custom patches locally, but is a hassle to maintain, so better to send them upstream.